### PR TITLE
Add Rect44 keyboard

### PR DESCRIPTION
**Pull request by koshinoya**
## Description
A 44-48 keys ANSI/ISO layout keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rect44/keymaps/ansi_44key/config.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2020-2021 koshinoya
```
Comment (zvecr) [inline keyboards/rect44/keymaps/ansi_44key/config.h:1]: This file should be removed

File: keyboards/rect44/keymaps/ansi_47key/config.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2020-2021 koshinoya
```
Comment (zvecr) [inline keyboards/rect44/keymaps/ansi_47key/config.h:1]: This file should be removed

File: keyboards/rect44/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2020-2021 koshinoya
```
Comment (zvecr) [inline keyboards/rect44/keymaps/default/config.h:1]: This file should be removed

File: keyboards/rect44/keymaps/iso_48key/config.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2020-2021 koshinoya
```
Comment (zvecr) [inline keyboards/rect44/keymaps/iso_48key/config.h:1]: This file should be removed

File: keyboards/rect44/readme.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+A 40% keyboard with various layouts.
+Includes hot-swap switch sockets and bended 1.5mm stainless sheet metal case.
+
+* Keyboard Maintainer: [koshinoya](https://github.com/koshinoya)
+* Hardware Supported: Rect44 PCB and case.
+* Hardware Availability: [BOOTH](https://koshinoya.booth.pm/items/1856336)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make rect44:ansi_44key
```
Comment (zvecr) [inline keyboards/rect44/readme.md:14]:
```suggestion
    make rect44:default

Flashing example for this keyboard:

    make rect44:default:flash
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/rect44/rules.mk [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/rect44/rules.mk:20]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/rect44/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Rect44
+
+![Rect44](https://i.imgur.com/TWXGnfm.jpg)
```
Comment (fauxpark) [inline keyboards/rect44/readme.md:3]:
```suggestion
![Rect44](https://i.imgur.com/TWXGnfml.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/rect44/config.h [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x4dd7
```
Comment (drashna) [inline keyboards/rect44/config.h:23]:
```suggestion
#define PRODUCT_ID      0x4DD7
```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by koshinoya** [comment]
Thanks everyone for review!

**Pull request merged**